arbor/fvm_layout.cpp

```diff
@@ -236,6 +236,62 @@ fvm_discretization fvm_discretize(const std::vector<mc_cell>& cells) {
     return D;
 }
 
+// Get vector of gap_junctions
+
+std::vector<gap_junction> fvm_gap_junctions(const std::vector<mc_cell>& cells, const std::vector<cell_gid_type>& gids,
+        const recipe& rec, const fvm_discretization& D) {
+    std::vector<gap_junction> v;
```

Comment:
As this function only operates on cell data post-discretization, we can move it to `fvm_lowered_cell_impl.hpp`. This way also, we can keep the discretization code one level removed from the recipe description — currently the two discretization routines just need the vector of `mc_cell` and the mechanism catalogue.